src/toil/jobStores/aws/jobStore.py

```diff
@@ -149,22 +154,46 @@ class AWSJobStore(AbstractJobStore):
     partitioned into chunks of 1024 bytes and each chunk is stored as a an attribute of the SDB
     item representing the job. UUIDs are used to identify jobs and files.
     """
-
     @classmethod
-    def loadOrCreateJobStore(cls, locator, config=None, **kwargs):
-        region, namePrefix = locator.split(':')
-        if not cls.bucketNameRe.match(namePrefix):
+    def _processLocator(cls, locator):
+        """
+        Processes the job store locator and returns an Expando object containing:
+            - region: The AWS region of the job store as a string.
+            - namePrefix: The name prefix of the job store as a string.
+
+        The syntax for an AWS job store locator string is as follows:
+            aws:<AWS region>:<name prefix>
+
+        For AWS regions see:
+            http://docs.aws.amazon.com/general/latest/gr/rande.html#s3_region
+            http://docs.aws.amazon.com/general/latest/gr/rande.html#sdb_region
+
+        For name prefix syntax see exception messages below.
+
+        :param str locator: The location of the job store.
+        :return: An Expando object containing info about the job store.
+        :rtype: bd2k.util.expando.Expando
+        :raises: ValueError
+        """
+        locator = locator.replace('aws:', '', 1)
```

Comment:
Do not like this because it's mixing split and replace. You can get by with just using split() and assert.